components/molecule/modal/src/index.js

```diff
@@ -31,152 +32,168 @@ const toggleWindowScroll = disableScroll => {
   window.document.body.classList.toggle('is-MoleculeModal-open', disableScroll)
 }
 
-const MoleculeModal = ({
-  children,
-  closeOnEscKeyDown = false,
-  closeOnOutsideClick = false,
-  enableContentScroll = false,
-  fitContent = false,
-  fitWindow = false,
-  floatingIconClose = false,
-  size,
-  header,
-  iconClose = false,
-  isClosing,
-  isOpen = false,
-  onAnimationEnd,
-  onClose = () => {},
-  portalContainerId = 'modal-react-portal',
-  usePortal = true,
-  withoutIndentation = false,
-  isContentless
-}) => {
-  const wrapperRef = useRef()
-
-  const [isClientReady, setIsClientReady] = useState(false)
-
-  const getContainer = () => {
-    let containerDOMEl = document.getElementById(portalContainerId)
-    if (!containerDOMEl) {
-      containerDOMEl = document.createElement('div')
-      containerDOMEl.id = portalContainerId
-      document.body.appendChild(containerDOMEl)
+const combineRefs = (...refs) => value => {
```

Comment:
wtf?

Reply:
What do you mean? Is about the function logic or the implementation?

Reply:
📞?

Reply:
in review

Reply:
Here I made an example indicating the formal proposed way 4 coding this using [that lib](https://github.com/thebuilder/react-intersection-observer) as described in the first FAQ of its readme. [https://codesandbox.io/s/condescending-pascal-kj3ff](https://codesandbox.io/s/condescending-pascal-kj3ff)

Reply:
I personally think its a clean solution to do that obtaining the same

Reply:
Mmm I have a couple of doubts fo the proposed implementation in the sandbox:
- It creates a ref **conditionally** with useRef, and it breaks a  rule of hook about using them conditionally. That's why I prefer the merging callback to check if the passed references exist or not.
- In case we'll need to use a similar approach for other components (which could happen often in a components library), we should duplicate this implementation if there are more references to merge, whether the ideal option could be to abstract this logic in a custom hook, something like this
```js
function combineRefs(...refs) {
  return function setReferences(node) {
    refs.forEach((ref) => {
      if (typeof ref === "function") ref(node);
      else if (ref !== null) ref.current = node;
    });
  };
}

export function useRefs(...refs) {
  return useCallback(combineRefs(...refs), refs);
}
```
so we'll have a utility hook that returns a memorized reference function to pass directly to the DOM element.

Reply:
take a look on actual implementation

Reply:
I've checked it, it's ok and should work for forwardedRef when it is not a function ref, in that case we still need to check for it's type to make it working in any possible scenario